### PR TITLE
feature:S3C-3540 Implement CreateS3Credentials and Test Cases

### PR DESCRIPTION
**Pull request by saitharunthodupunuru**
1.  Implement CreateS3Credentials and Test Cases
2. Adding extra logs

### CODE REVIEW

File: osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java [hunk 1]
```diff
@@ -304,7 +308,24 @@ public PageOfUsers queryUsers(long offset, long limit, String filter) {
...
+        try {
+            logger.info("Create S3 Credential request received:: tenant ID:{}, user ID:{}",
+                    tenantId, userId);
+
+            Credentials tempCredentials = getCredentials(tenantId);
+            final AmazonIdentityManagement iamClient = vaultAdmin.getIAMClient(tempCredentials, appEnv.getRegionInfo().get(0));
+
+            OsisS3Credential credential =  createOsisCredential(tenantId, userId, null, null, iamClient);
+
+            logger.trace("Create S3 Credential response with secret:{}", new Gson().toJson(credential));
```
Comment (anurag4DSB) [inline osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java:320]: I researched about this, we should not log secret key even in trace logs.

More info in the summary of this ticket: https://scality.atlassian.net/browse/S3C-4366

**Review by anurag4DSB — CHANGES_REQUESTED** [review]

File: osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java (hunk 1 above)
Comment (anurag4DSB) [reply osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java:320]: As discussed for older logs(already committed), we can clean up later.

Comment (saitharunthodupunuru) [reply osis-core/src/main/java/com/scality/osis/service/impl/ScalityOsisService.java]: fixed in f250542fa2138cb24ab0e2fd1ceeb97abef1750d

**Review by anurag4DSB — APPROVED** [review]

**Review by ilkescality — APPROVED** [review]

**Comment by saitharunthodupunuru** [comment]
Squash merged the commits

**Pull request merged**